### ISSUE TITLE
No method in multimethod 'connection-details->spec' for dispatch value: :snowflake

### ISSUE THREAD

**Issue by jqtruong** [metabase/metabase#12426]
**Describe the bug**
During the add data step of the setup, i entered my snowflake config but i get an error. The logs show a really long stack trace. (Also see **Update 2** below).

**Logs**
It starts with:
```
[backend] 04-29 14:30:06 INFO driver.impl :: Initializing driver :snowflake...
[backend] 04-29 14:30:06 DEBUG plugins.init-steps :: Loading plugin namespace metabase.driver.snowflake...
[backend] 04-29 14:30:06 ERROR driver.util :: Database connection error
[backend] clojure.lang.ExceptionInfo: Could not locate metabase/driver/snowflake__init.class, metabase/driver/snowflake.clj or metabase/driver/snowflake.cljc on classpath. {:classloader #object[clojure.lang.DynamicClassLoader 0x4768b95c "clojure.lang.DynamicClassLoader@4768b95c"], :classpath-urls ()}
```

It goes on for a while, and ends with:

```
[backend] 	... 1 more
[backend] 04-29 14:30:06 DEBUG middleware.log :: POST /api/setup/validate 400 5.6 ms (0 DB calls) 
[backend] {:errors {:dbname "No method in multimethod 'connection-details->spec' for dispatch value: :snowflake"}}
[backend] 
```

**To Reproduce**
1. check out the [release 0.35.x branch](https://github.com/metabase/metabase/tree/release-0.35.x)
1. ```$ ./bin/build no-translations```
1. ```$ yarn dev```
1. go through the Metabase admin setup and when you reach Add your data...
1. go through the Snowflake db form and submit

**Expected behavior**
Less shock and awe, more Snowflake db added.

**Screenshots**
![image](https://user-images.githubusercontent.com/308746/80644260-f8788c80-8a2e-11ea-99d8-17e83fbca23e.png)

**Information about your Metabase Installation:**
```
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.129 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "Java(TM) SE Runtime Environment",
    "java.runtime.version": "11.0.5+10-LTS",
    "java.vendor": "Oracle Corporation",
    "java.vendor.url": "http://java.oracle.com/",
    "java.version": "11.0.5",
    "java.vm.name": "Java HotSpot(TM) 64-Bit Server VM",
    "java.vm.version": "11.0.5+10-LTS",
    "os.name": "Mac OS X",
    "os.version": "10.15.4",
    "user.language": "en",
    "user.timezone": "America/Chicago"
  },
  "metabase-info": {
    "databases": [
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "h2",
    "application-database-details": {
      "database": {
        "name": "H2",
        "version": "1.4.197 (2018-03-18)"
      },
      "jdbc-driver": {
        "name": "H2 JDBC Driver",
        "version": "1.4.197 (2018-03-18)"
      }
    },
    "run-mode": "dev",
    "version": {
      "tag": "v0.35.3",
      "hash": "e31ad3d",
      "branch": "release-0.35.x",
      "date": "2020-04-28"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

**Severity**
Blocked trying upgrade from 31 to 35.

**Additional context**
https://discourse.metabase.com/t/no-method-in-multimethod-connection-details-spec-for-dispatch-value-snowflake/9773/13

**Update 1**: using Leiningen 2.9.1 on Java 11.0.5 Java HotSpot(TM) 64-Bit Server VM

**Update 2**: if i run 
```
java -jar /path/to/metabase/target/uberjar/metabase.jar
```
from outside of that `/path/to/metabase` directory, no issue with the Snowflake driver.

**Comment by jqtruong** [comment]
This does not happen with the official release https://www.metabase.com/start/jar.html.

**Comment by jqtruong** [comment]
here is the build logs for the Snowflake driver:

```
Checksum has changed.
Verifying snowflake driver...
Checking whether resources/modules/snowflake.metabase-driver.jar exists....
File exists.
Checking whether driver contains main class file metabase/driver/snowflake__init.class...
Main class file found.
Checking whether driver contains plugin manifest...
Plugin manifest found.
Driver verification successful.
Deleting old versions of snowflake driver...
metabase-core already installed to local Maven repo.
Metabase uberjar already built.
Building parent drivers (if needed)...
Building snowflake driver...
Applying task uberjar to []
Applying task javac to nil
Running javac with [-target 1.8 -source 1.8 @/var/folders/6g/g303w8990bs2ddnnjvm9f5d80000gp/T/.leiningen-cmdline10430121795908838797.tmp]
Applying task compile to nil
04-30 15:35:11 DEBUG plugins.classloader :: Using NEWLY CREATED classloader as shared context classloader: clojure.lang.DynamicClassLoader@64440065
Created /Users/jerome.truong/Applications/metabase/modules/drivers/snowflake/target/uberjar+provided/snowflake-driver-1.0.0-SNAPSHOT-3.12.2.jar
Including snowflake-driver-1.0.0-SNAPSHOT-3.12.2.jar
Including snowflake-jdbc-3.12.2.jar
Created /Users/jerome.truong/Applications/metabase/modules/drivers/snowflake/target/uberjar/snowflake.metabase-driver.jar
04-30 15:35:33 DEBUG plugins.classloader :: Using Clojure base loader as shared context classloader: clojure.lang.DynamicClassLoader@3bda1f0
04-30 15:35:33 INFO metabase.util :: Loading Metabase...
04-30 15:35:33 INFO metabase.util :: Maximum memory available to JVM: 4.0 GB
Stripping duplicate classes and extra-compressing /Users/jerome.truong/Applications/metabase/modules/drivers/snowflake/target/uberjar/snowflake.metabase-driver.jar...
Done. wrote: 18041 skipped: 36
Original size: 26.2 MB
Stripped/extra-compressed size: 26.0 MB
Copying to /Users/jerome.truong/Applications/metabase/modules/drivers/snowflake/target/uberjar/snowflake.metabase-driver.jar...
Copying /Users/jerome.truong/Applications/metabase/modules/drivers/snowflake/target/uberjar/snowflake.metabase-driver.jar -> /Users/jerome.truong/Applications/metabase/resources/modules/snowflake.metabase-driver.jar
Verifying snowflake driver...
Checking whether resources/modules/snowflake.metabase-driver.jar exists....
File exists.
Checking whether driver contains main class file metabase/driver/snowflake__init.class...
Main class file found.
Checking whether driver contains plugin manifest...
Plugin manifest found.
Driver verification successful.
Saving checksum for source files to /Users/jerome.truong/Applications/metabase/modules/drivers/snowflake/target/checksum.txt
```
it doesn't seem like i cleaned properly, but i just wanted to build a single driver so perhaps the top of the log can be ignored. What matters is that "Main class file found" but the error i got while running local, with `yarn dev`, i got this: ```[backend] clojure.lang.ExceptionInfo: Could not locate metabase/driver/snowflake__init.class, metabase/driver/snowflake.clj or metabase/driver/snowflake.cljc on classpath. {:classloader #object[clojure.lang.DynamicClassLoader 0x4768b95c "clojure.lang.DynamicClassLoader@4768b95c"], :classpath-urls ()}```

**Comment by jqtruong** [comment]
oh, it's indeed not in the specified folder:

```
[15:44:49 | jerome.truong@jtruong-LVCF metabase-release-0.35.x]
$ ls -l metabase/driver/ | grep __init.class
-rw-r--r--    1 jerome.truong  wheel  17353 Apr 29 19:43 common__init.class
-rw-r--r--    1 jerome.truong  wheel  18032 Apr 29 19:43 h2__init.class
-rw-r--r--    1 jerome.truong  wheel   8608 Apr 29 19:43 impl__init.class
-rw-r--r--    1 jerome.truong  wheel  13930 Apr 29 19:43 mysql__init.class
-rw-r--r--    1 jerome.truong  wheel  18815 Apr 29 19:43 postgres__init.class
-rw-r--r--    1 jerome.truong  wheel   4797 Apr 29 19:43 sql__init.class
-rw-r--r--    1 jerome.truong  wheel   5661 Apr 29 19:43 sql_jdbc__init.class
-rw-r--r--    1 jerome.truong  wheel   7597 Apr 29 19:43 util__init.class
```

**Comment by flamber** [comment]
This seems to have been solved by #12436 - from the [forum comment](https://discourse.metabase.com/t/no-method-in-multimethod-connection-details-spec-for-dispatch-value-snowflake/9773/24?u=flamber):
> running `yarn dev` from the `master` branch does in fact solve the missing driver issue. I did not quite understand the concept of profiles, but i did not notice an infinite loop either
